apacheconfig/wloader.py

```diff
@@ -174,3 +289,87 @@ def __str__(self):
                 % (self.__class__.__name__,
                    str([self._type] +
                        [_restore_original(word) for word in self._raw])))
+
+
+class BlockNode(ListNode):
+    """Creates object containing data for a block.
+
+    Manages any preceding whitespace before the opening block tag, and
+    corresponding :class:`apacheconfig.ListNode` methods will manipulate
+    block contents.
+    """
+
+    def __init__(self, raw, parser):
+        self._whitespace = ""
+        self._type = raw[0]
+        start = 1
+        if isinstance(raw[start], str) and raw[start].isspace():
+            self._whitespace = raw[start]
+            start += 1
+        self._full_tag = LeafASTNode(('statement',) + raw[start])
+        self._close_tag = raw[-1]
+        self._contents = None
+        if len(raw[start + 1]) > 0:
+            super(BlockNode, self).__init__(raw[start + 1], parser)
+
+    @classmethod
+    def parse(cls, raw_str, parser):
+        """Factory for :class:`apacheconfig.BlockNode` from a config string.
+
+        Args:
+            raw_str (string): The text to parse.
+            parser (:class:`apacheconfig.ApacheConfigParser`): parser object
+                to use.
+        Returns:
+            :class:`apacheconfig.BlockNode` containing metadata parsed from
+            ``raw_str``.
+        """
+        raw = parser.parse(raw_str)
+        return cls(raw[1], parser)
+
+    @property
+    def tag(self):
+        """Returns tag name for this block as a string.
+
+        For instance, ``<block details>\\n</block>`` has tag ``"block"``.
+        """
+        return self._full_tag.name
+
+    @property
+    def arguments(self):
```

Comment:
I wonder what's the conventional term for block contents? Is it `argument` in Apache parlance? If not, may be we could borrow something from beloved XML? Something familiar to fellow programmers...

Reply:
Good question! Just to be explicit, this is about `arguments` in a block like
```
<Directive arguments>
((contents))
</Directive>
```

The Apache docs refer to them as parameters or [arguments](https://httpd.apache.org/docs/2.4/mod/core.html#directory). I'll write this out in the class docstring, too.